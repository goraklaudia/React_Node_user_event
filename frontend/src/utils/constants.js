export const messages = {
    USER_ADDED: 'User successfully added.',
}

export const URL_SERVER = 'http://localhost:4000/api'