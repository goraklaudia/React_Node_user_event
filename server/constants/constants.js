export const constants = {
    ERROR: 'Something went wrong.',
    SUCCESS: 'Success.',
    USER_ADDED: 'User successfully added.',
    MISSING_PARAMS: 'Some parameters are missing.'
}

export const resStatuses = {
    [constants.SUCCESS]: 200,
    [constants.ERROR]: 500,
    [constants.USER_ADDED]: 201,
    [constants.MISSING_PARAMS]: 400
}
