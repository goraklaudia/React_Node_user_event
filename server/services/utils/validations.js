export const validateEmail = (email) => {
    return /^$|.+@.+..+/.test(email);
}

export const validateDate = (date) => {
    return /^\d{4}\-(0?[1-9]|1[012])\-(0?[1-9]|[12][0-9]|3[01])$/.test(date);
}