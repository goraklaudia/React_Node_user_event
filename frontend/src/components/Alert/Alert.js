import { Alert as AlertForm, Snackbar } from "@mui/material";
import { messages } from '../../utils/constants';

const Alert = ({ msg, open, setOpen }) => {
    const generateSeverity = () => {
        return msg === messages.USER_ADDED ? 'success' : 'error';
    }

    const handleClose = (e, reason) => {
        if (reason === 'clickaway') {
            return;
        }
        return setOpen({ msg: '', open: false });
    }

    return (
        <Snackbar
            open={open}
            autoHideDuration={5000}
            onClose={handleClose}
            anchorOrigin={{
                vertical: 'top',
                horizontal: 'center'
            }}
        >
            <AlertForm
                onClose={handleClose}
                severity={generateSeverity()}
            >
                {msg}
            </AlertForm>
        </Snackbar>
    );
};

export default Alert;
