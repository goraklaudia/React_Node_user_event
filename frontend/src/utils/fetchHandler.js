import { URL_SERVER } from './constants';

export const fetchHandler = async (endpoint, options, setAlert, successCallback) => {
    if (endpoint && options && setAlert) {
        fetch(`${URL_SERVER}${endpoint}`, options)
            .then((res) => res.json())
            .then((res) => {
                setAlert({ msg: res.msg, open: true });
                successCallback();
            })
            .catch((err) => {
                setAlert({ msg: 'Error during request', open: true });
            })
    }
}