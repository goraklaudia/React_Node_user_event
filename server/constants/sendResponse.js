import { resStatuses } from './constants.js';

const sendResponse = (res, payload) => {
    if (typeof payload === 'string') {
        res.status(resStatuses[payload] || 200).json({ msg: payload });
    } else {
        const { result, ...data } = payload;
        res.status(resStatuses[result]).json({ msg: result, ...data });
    }
}

export default sendResponse;