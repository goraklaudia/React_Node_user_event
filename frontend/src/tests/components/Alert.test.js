import React from 'react';
import { act, create } from 'react-test-renderer';
import Alert from '../../components/Alert/Alert';
import { messages } from '../../utils/constants';

describe('Component Alert', () => {
    let alertComponent = {};
    let setOpen;

    beforeAll(() => {
        setOpen = jest.fn();
        alertComponent = create(<Alert msg={messages.USER_ADDED} open={false} setOpen={setOpen} />);
    });

    it('should render correctly', () => {
        expect(alertComponent.toJSON()).toMatchSnapshot();
    });

    it('should update content of the message correctly', () => {
        act(() => {
            alertComponent.update(<Alert msg={'Error'} open={false} setOpen={setOpen} />);
        })

        expect(alertComponent.toJSON()).toMatchSnapshot();
    });
})