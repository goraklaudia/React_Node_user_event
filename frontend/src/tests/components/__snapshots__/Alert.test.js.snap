// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Alert should render correctly 1`] = `null`;

exports[`Component Alert should update content of the message correctly 1`] = `null`;
